### PR TITLE
feat: sql runner table fields

### PR DESCRIPTION
**Pull request by joaoviana**
Closes: #10551 

### Description:

- uses new type `ApiWarehouseTableFields` when getting table fields 
- adds `react-resizable` so we can resize the table list field
- adds `TableFields` component 

(will add search feature on: https://github.com/lightdash/lightdash/issues/10569)
(will address stable sidebar-collapsing animation on: https://github.com/lightdash/lightdash/issues/10571)

Demo: 

https://github.com/lightdash/lightdash/assets/7611706/9c9ed044-7b00-493b-8545-556a957fb5c4

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/backend/src/services/ProjectService/ProjectService.ts [hunk 1]
```diff
@@ -2657,15 +2658,23 @@ export class ProjectService extends BaseService {
...
-        return warehouseCatalog;
+        if (!schema) {
+            throw new NotFoundError(
+                'Schema not found in warehouse credentials',
+            );
+        }
+
+        return warehouseCatalog[warehouseClient.credentials.type][schema][
+            tableName
+        ];
```
Comment (joaoviana) [inline packages/backend/src/services/ProjectService/ProjectService.ts:2677]: I would like to return only an object with the fields for a table. Would this be the correct way of doing this? 
Do you think we shouldn't  return this format instead?
If we should, what do you think about the handling of undefined `schema`?

File: packages/e2e/cypress/e2e/api/sqlRunner.cy.ts [hunk 2]
```diff
@@ -64,32 +64,23 @@ Object.entries(warehouseConnections).forEach(
...
-                    const schemaResults = databaseResults[schema];
-                    expect(Object.keys(schemaResults)).to.include('orders');
-                    expect(Object.keys(schemaResults)).to.not.include(
-                        'customers',
-                    );
-                    const fieldResults = schemaResults.orders;
+                    const { results } = resp.body;
 
-                    expect(Object.keys(fieldResults)).to.have.length.gt(5);
+                    expect(Object.keys(results)).to.have.length.gt(5);
```
Comment (joaoviana) [inline packages/e2e/cypress/e2e/api/sqlRunner.cy.ts:76]: now we just return an object with the table fields.

File: packages/frontend/src/features/sqlRunner/hooks/useTableFields.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    return useQuery<ApiWarehouseTableFields, ApiError>({
+        queryKey: ['sqlRunner', 'tables', tableName, projectUuid],
+        queryFn: () =>
+            fetchTableFields({
+                projectUuid,
+                tableName,
+            }),
+        retry: false,
+        enabled: !!tableName,
+        // TODO: handle error on no schema found
```
Comment (joaoviana) [inline packages/frontend/src/features/sqlRunner/hooks/useTableFields.tsx:33]: if the handling of `undefined` schema is OK, I think we should handle it at some point here...?

Comment (rephus) [reply packages/frontend/src/features/sqlRunner/hooks/useTableFields.tsx:33]: afaik the schema is needed to get the tables first, in the backend, so you might not reach this state if that happens...

File: packages/backend/src/services/ProjectService/ProjectService.ts [hunk 4]
```diff
@@ -2656,16 +2676,28 @@ export class ProjectService extends BaseService {
...
+        const warehouseCatalog = await warehouseClient.getFields(
             tableName,
             schema,
             queryTags,
         );
 
         await sshTunnel.disconnect();
 
-        return warehouseCatalog;
+        return warehouseCatalog[database][schema][tableName];
```
Comment (joaoviana) [inline packages/backend/src/services/ProjectService/ProjectService.ts:2700]: now we get the `database` depending on the warehouse. 
LMK what you think @rephus and if you would do it differently.

**Review by rephus — APPROVED** [review]
Looks good to me

File: packages/frontend/src/features/sqlRunner/hooks/useTableFields.tsx (hunk 3 above)
Comment (joaoviana) [reply packages/frontend/src/features/sqlRunner/hooks/useTableFields.tsx:33]: removed comment

**Pull request merged**